README.rst
lizard-registration
==========================================

Introduction

Provides the functionalities to manage user for vss.

Usage
-------------------------------------------
To use this app you need to include it into vss project.


CHANGES.rst
Changelog of lizard-registration
===================================================


0.4.4 (unreleased)
------------------

- Nothing changed yet.


0.4.3 (2013-01-08)
------------------

- Login by ip pp415.


0.4.2 (2012-10-01)
------------------

- Fixed syntax errors

- Ordered users by username.


0.4.1 (2012-06-21)
------------------

- Fixed pylint errors.

- Fixed auto-login functionality by forwarded ip-address.

- Fixed error on get_user_permissions_overall for anonymous user.


0.4 (2012-06-20)
----------------

- Fixed auto-login as 'raadpleger' functionality.

0.3 (2012-06-15)
----------------

- Changed label of fields and buttons, fixed messages in templates, forms.


0.2 (2012-04-10)
----------------

- Added comments.

- Removed cancel button from users_table_view.


0.1.7 (2012-04-02)
------------------

- Bugfix sessioncontextstore save for iprange logins.


0.1.6 (2012-03-30)
------------------

- Fixed pylint errors.


0.1.5 (2012-03-30)
------------------

- Updated functionality to view user_groups on update user.


0.1.4 (2012-03-30)
------------------

- Implemented functionality to add 'raadpleger' user_group to a new user.

- Implemented 'helpdesk' user_group.

- Added some comments.


0.1.3 (2012-03-12)
------------------

- Fixes spelling error in user management screen (fixes Projectplace issue 92).


0.1.2 (2012-02-27)
------------------

- added a context store, including model and api

- added function for getting all permissions of a user (including through lizard-security)


0.1.1 (2012-02-08)
------------------

- Added initial migration schema.


0.1 (2012-02-07)
----------------

- Initial library skeleton created by nensskel.  [Alexandr Seleznev]

- Moved 'Gebruikersbeheer' functionalities from vss into
  lizard_registration (issue 1).



lizard_registration/migrations/0002_auto__add_sessioncontextstore__add_usercontextstore.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Adding model 'SessionContextStore'
        db.create_table('lizard_registration_sessioncontextstore', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('user', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['auth.User'])),
            ('session_key', self.gf('django.db.models.fields.CharField')(max_length=40, unique=True, null=True, blank=True)),
            ('context', self.gf('django.db.models.fields.TextField')(null=True, blank=True)),
        ))
        db.send_create_signal('lizard_registration', ['SessionContextStore'])

        # Adding model 'UserContextStore'
        db.create_table('lizard_registration_usercontextstore', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('user', self.gf('django.db.models.fields.related.OneToOneField')(to=orm['auth.User'], unique=True)),
            ('context', self.gf('django.db.models.fields.TextField')(null=True, blank=True)),
        ))
        db.send_create_signal('lizard_registration', ['UserContextStore'])


    def backwards(self, orm):
        
        # Deleting model 'SessionContextStore'
        db.delete_table('lizard_registration_sessioncontextstore')

        # Deleting model 'UserContextStore'
        db.delete_table('lizard_registration_usercontextstore')


    models = {
        'auth.group': {
            'Meta': {'object_name': 'Group'},
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '80'}),
            'permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'})
        },
        'auth.permission': {
            'Meta': {'ordering': "('content_type__app_label', 'content_type__model', 'codename')", 'unique_together': "(('content_type', 'codename'),)", 'object_name': 'Permission'},
            'codename': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'content_type': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['contenttypes.ContentType']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '50'})
        },
        'auth.user': {
            'Meta': {'object_name': 'User'},
            'date_joined': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'email': ('django.db.models.fields.EmailField', [], {'max_length': '75', 'blank': 'True'}),
            'first_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'groups': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Group']", 'symmetrical': 'False', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'is_active': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'is_staff': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'is_superuser': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'last_login': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'last_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'password': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'user_permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'}),
            'username': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '30'})
        },
        'contenttypes.contenttype': {
            'Meta': {'ordering': "('name',)", 'unique_together': "(('app_label', 'model'),)", 'object_name': 'ContentType', 'db_table': "'django_content_type'"},
            'app_label': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'model': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        'lizard_registration.iprangelogin': {
            'Meta': {'ordering': "['user', 'ipadres']", 'object_name': 'IPrangeLogin'},
            'created_on': ('django.db.models.fields.DateField', [], {'auto_now': 'True', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'ipadres': ('django.db.models.fields.IPAddressField', [], {'max_length': '15'}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['auth.User']"})
        },
        'lizard_registration.organisation': {
            'Meta': {'object_name': 'Organisation'},
            'description': ('django.db.models.fields.CharField', [], {'max_length': '256', 'null': 'True', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '128'})
        },
        'lizard_registration.sessioncontextstore': {
            'Meta': {'object_name': 'SessionContextStore'},
            'context': ('django.db.models.fields.TextField', [], {'null': 'True', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'session_key': ('django.db.models.fields.CharField', [], {'max_length': '40', 'unique': 'True', 'null': 'True', 'blank': 'True'}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['auth.User']"})
        },
        'lizard_registration.usercontextstore': {
            'Meta': {'object_name': 'UserContextStore'},
            'context': ('django.db.models.fields.TextField', [], {'null': 'True', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'user': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['auth.User']", 'unique': 'True'})
        },
        'lizard_registration.userprofile': {
            'Meta': {'object_name': 'UserProfile'},
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'organisation': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['lizard_registration.Organisation']"}),
            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['auth.User']", 'unique': 'True'})
        }
    }

    complete_apps = ['lizard_registration']
